File: Backend/Newsletter/Data/SearchRequests/ArticleSearchRequest.cs
namespace Newsletter.Data.SearchRequests {
    public class ArticleSearchRequest : SearchRequestBase {
        public Guid? OrganizationId { get; set; }
        public Guid? NewsletterId { get; set; }
        public Guid? CreatedById { get; set; }
        public bool? Published { get; set; } = true;
        public DateTime? From { get; set; }
        public DateTime? To { get; set; }
    }
}

File: Frontend/src/app/api/services/models/article-search-request.ts
import { SearchRequestBase } from './search-request-base';

export class ArticleSearchRequest extends SearchRequestBase {
  organizationId?: string;
  newsletterId?: string;
  createdById?: string;
  published?: boolean = false;
  from?: Date;
  to?: Date;

  public override reset(): void {
    this.searchTerm = '';
    this.skip = 0;
    this.take = 50;
    this.organizationId = undefined;
    this.newsletterId = undefined;
    this.createdById = undefined;
    this.published = true;
    this.from = undefined;
    this.to = undefined;
  }
}
